sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/meta/provider/kafka/KafkaTableProvider.java

```diff
@@ -85,6 +89,33 @@ public BeamSqlTable buildBeamSqlTable(Table table) {
         } catch (ClassNotFoundException e) {
           throw new IllegalArgumentException("Incorrect proto class provided: " + protoClassName);
         }
+      case THRIFT:
+        final String thriftClassName = properties.getString("thriftClass");
```

Comment:
I'd delegate this part of code to a function. It'll get a bit messy after we add another case to the switch statement.

Reply:
Everything else was inlined here, so I thought I'd keep it "consistent".
But you're right, thrift also requires a protocol spec, which makes it even more involved than the proto case, so it probably deserves its own method.

Reply:
I agree that at least PROTO case should be delegated to a method as well

Reply:
I'll extract that one too